// async : this function will return the promise in any situation wherever the async function is used.
/*
async function greet() {
    // throw("This error is self gererated!!!");
    return "Hello!!!";
}

// let promiseMs = greet();
// console.log(promiseMs);// this will print undefined

greet()
    .then((result) => {
        console.log("Promise was resolved");
        console.log("Result was :",result);
    })
    .catch((error) => {
        console.log("Promise was rejected with error :", error);
    })
*/





/*
// making arrow function async >>>
let demo = async () => {
    return 5;
}
console.log(demo());
*/
